packages/ui/src/composites/wui-tabs/index.ts

```diff
@@ -13,36 +14,90 @@ export class WuiTabs extends LitElement {
 
   @property() public onTabChange: (index: number) => void = () => null
 
+  @property({ type: Array }) public buttons: HTMLButtonElement[] = []
+
   @state() public activeTab = 0
 
+  @state() public localTabWidth = '100px'
+
+  @state() public isDense = false
+
   // -- Render -------------------------------------------- //
   public override render() {
-    const isDense = this.tabs.length > 3
+    this.isDense = this.tabs.length > 3
 
     this.style.cssText = `
       --local-tab: ${this.activeTab};
-      --local-tab-width: ${isDense ? '80px' : '100px'}
+      --local-tab-width: ${this.localTabWidth};
+      --local-dense-tab-width: max-content;
     `
 
+    this.dataset['type'] = this.isDense ? 'flex' : 'block'
+
     return this.tabs.map((tab, index) => {
       const isActive = index === this.activeTab
 
       return html`
-        <button @click=${() => this.onTabClick(index)} data-active=${isActive}>
+        <button
+          style="{{width:
+          20px}}"
+          @click=${() => this.onTabClick(index)}
+          data-active=${isActive}
+        >
           <wui-icon size="sm" color="inherit" name=${tab.icon}></wui-icon>
-          <wui-text variant=${isDense ? 'tiny-600' : 'small-600'} color="inherit">
-            ${tab.label}
-          </wui-text>
+          <wui-text variant="small-600" color="inherit"> ${tab.label}</wui-text>
         </button>
       `
     })
   }
 
+  override firstUpdated() {
```

Comment:
Could you explain what's going on here / why is this needed?

Reply:
After a component has been mounted, we save button elements in an array. Then, we call a function to run the initial animation so that the first tab is active and visible. 

This is necessary because by default, all the text components from the buttons are absolutely positioned. We need to calculate the width that the active button should have.